flair/rpc/rpc.go

```diff
@@ -616,15 +616,20 @@ func (s *server) List(ctx context.Context, req *ppb.ListRequest) (*ppb.ListRespo
 				ar.Replicas = 1
 			}
 			if existing, present := ars[ar.Hash]; !present {
-				resp.ActionResults = append(resp.ActionResults, ar)
 				ars[ar.Hash] = ar
 			} else {
-				if existing.LastAccessed < ar.LastAccessed {
-					existing.LastAccessed = ar.LastAccessed
+				// Update ars with the replica, and use the more recent lastAccessed time.
+				ar.Replicas += existing.Replicas
+				if ar.LastAccessed < existing.LastAccessed {
+					ar.LastAccessed = existing.LastAccessed
 				}
-				existing.Replicas += ar.Replicas
+				ars[ar.Hash] = ar
```

Comment:
Hmmm so I think this was supposed to work by the action results in the map being pointers, so updating them in-place should still be reflected without us needing to re-set them in the map.

Maybe I'm missing something though?